CMakeLists.txt

```diff
@@ -1,11 +1,144 @@
+add_library(readerwriterqueue INTERFACE)
 
-install(FILES atomicops.h readerwriterqueue.h readerwritercircularbuffer.h LICENSE.md
+# This needs to be updated everytime the library tag version updates. The version is not 1.0.0, it's 1.0.6 according to tags.
+set_target_properties(readerwriterqueue PROPERTIES
+        SOVERSION 1
+        VERSION 1.0.6)
+
+#see this talk for why these changes need to be made https://www.youtube.com/watch?v=m0DwB4OvDXk
+
+target_include_directories(readerwriterqueue INTERFACE
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}>
+)
+
+set_target_properties(readerwriterqueue PROPERTIES EXPORT_NAME readerwriterqueue)
+
+# This is here to ensure parity with the install interface, but won't hide the previous target name.
+add_library(moodycamel::readerwriterqueue ALIAS readerwriterqueue)
+
+
+
+
+install(TARGETS readerwriterqueue
+        EXPORT readerwriterqueue_Targets
+        COMPONENT readerwriterqueue_Development
+        INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        COMPONENT creaderwriterqueue_RunTime
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        COMPONENT readerwriterqueue_RunTIme
+        NAMELINK_COMPONENT readerwriterqueue_Development
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        COMPONENT readerwriterqueue_Development
+)
+
+set(readerwriterqueue_INSTALL_CMAKEDIR
+        ${CMAKE_INSTALL_DATADIR}/readerwriterqueue
+        CACHE STRING "Path to readerwriterqueue cmake files"
+)
+
+install(EXPORT readerwriterqueue_Targets
+        DESTINATION ${readerwriterqueue_INSTALL_CMAKEDIR}
+        NAMESPACE moodycamel::
+        FILE readerwriterqueueTargets.cmake
+        COMPONENT readerwriterqueue_Development
+)
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/config.cmake.in
+        "${CMAKE_CURRENT_BINARY_DIR}/readerwriterqueueConfig.cmake"
+        INSTALL_DESTINATION ${readerwriterqueue_INSTALL_CMAKEDIR}
+)
+
+
+get_target_property(readerwriterqueue_VERSION readerwriterqueue VERSION)
+
+write_basic_package_version_file(readerwriterqueueConfigVersion.cmake VERSION ${readerwriterqueue_VERSION} COMPATIBILITY SameMajorVersion)
+
+install(FILES
+        ${CMAKE_CURRENT_BINARY_DIR}/readerwriterqueueConfig.cmake
+        ${CMAKE_CURRENT_BINARY_DIR}/readerwriterqueueConfigVersion.cmake
+        DESTINATION ${readerwriterqueue_INSTALL_CMAKEDIR}
+)
+
+
+install(FILES
+        atomicops.h
+        readerwriterqueue.h
+        readerwritercircularbuffer.h
+        LICENSE.md
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
+
+
+option(MOODYCAMEL_READERWRITERQUEUE_ENABLE_TESTS "Enables readerwriterqueue tests" OFF)
+
+if(${MOODYCAMEL_READERWRITERQUEUE_ENABLE_TESTS})
+
+    find_package(Threads REQUIRED)
```

Comment:
What is `Threads`?

Reply:
Threads is an imported target (ie a target not native to CMake which other CMake code creates an interface for) required to use std::thread in some contexts, it isn't something you have to install (unless you don't have pthreads or what ever the os specific threading library you used is), but is a CMake target CMake automatically creates for each system it supports.  see https://stackoverflow.com/questions/67912839/how-do-you-specify-a-threads-dependency-in-cmake-for-distributing-a-header-only.  You want to do this even on systems that don't require it, because CMake will do the right thing and do whatever the platform deems necessary for std::threads usage here.  On some systems it does nothing.

Reply:
Thanks. I find it interesting that a part of the standard library requires a special package.